### DISCUSSION TITLE
I have many routes that need to use "page" and "page_size". Is there a solution with class inheritance for Query ?

### DISCUSSION THREAD

**Discussion by Robin528919** [fastapi/fastapi#11813] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

I have many routes that need to use "page" and "page_size". Is there a solution with class inheritance for Query ?

**Comment by Robin528919** [comment]
```
@router.get("/item1")
def item1(
        page: int = 1,
        size: int = 10):
    pass

@router.get("/item2")
def item2(
        page: int = 1,
        size: int = 10):
    pass

@router.get("/item3")
def item3(
        page: int = 1,
        size: int = 10):
    pass

@router.get("/item4")
def item4(
        page: int = 1,
        size: int = 10):
    pass

@router.get("/item5")
def item5(
        page: int = 1,
        size: int = 10):
    pass
```    
  Is there a simpler way to solve the repeated parameters?

**Comment by YuriiMotov** [comment, marked as answer]
Format you code with the code block, please.

You can use dependency function for this reason:
```
async def get_page_size(self, page: int = 1, size: int = 10):
    return {"page": page, "size": size}

@router.get("/item1")
def item1(page_size: Annotated[dict[str, int], Depends(get_page_size)]):
    print(page_size["page"])
    print(page_size["size"])

```

**Reply by Robin528919** [reply]
class QueryBase(BaseModel):
    page: Optional[int] = Field(default=1, description="页面")
    page_size: Optional[int] = Field(default=30, description="查询数量")

    @validator('page', pre=True, always=True)
    def page_validator(cls, v):
        if v is None or v == '':
            return 1
        if isinstance(v, int) and v < 1:
            return 1
        if not isinstance(v, int):
            raise ValueError('page 必须是一个整数')
        return v

    @validator('page_size', pre=True, always=True)
    def page_size_validator(cls, v):
        if v is None or v == '':
            return 30
        if isinstance(v, int):
            if v < 1:
                raise ValueError('page_size 必须大于0')
            if v > settings.MAX_PAGE_SIZE:
                return settings.MAX_PAGE_SIZE
        else:
            raise ValueError('page_size 必须是一个整数')
        return v


async def query_page(
        page: Optional[int] = Query(default=1, description="页面"),
        page_size: Optional[int] = Query(default=30, description="查询数量")
):
    return QueryBase(page=page, page_size=page_size)


CommonsQuery = Annotated[QueryBase, Depends(query_page)]


# 查询任务列表
class QueryTaskList(QueryBase):
    task_type: int = Field(default=None, description=TaskType.descriptions())
    task_status: int = Field(default=None, description=TaskStatus.descriptions())
    user_id: Optional[int] = Field(default=None, description="账号分组id")


Is there a class inheritance scheme?
Is there a class inheritance scheme?

Is there a class inheritance scheme?
